go/dpagg/mean.go

```diff
@@ -213,6 +215,101 @@ func (bm *BoundedMeanFloat64) Result() float64 {
 	return clamped
 }
 
+// ComputeConfidenceInterval computes a confidence interval that contains the true mean with
+// probability greater than or equal to 1 - alpha. The computation is based exclusively on the
+// confidence intervals for sum and count. Thus no privacy budget is consumed by this operation.
+//
+// Result() needs to be called before ComputeConfidenceInterval, otherwise this will return an error.
+func (bm *BoundedMeanFloat64) ComputeConfidenceInterval(alpha float64) (noise.ConfidenceInterval, error) {
+	if !bm.resultReturned {
+		return noise.ConfidenceInterval{}, fmt.Errorf("Result() must be called before calling ComputeConfidenceInterval()")
+	}
+	// Brute force optimization for finding the tightest confidence interval: Iterate
+	// through a fixed numSplits number of splits for countAlpha where it takes the
+	// range of values strictly from 0 to alpha.
+	increment := alpha / float64(numSplits)
+	tightest := math.Inf(1)
+	var tightestConfInt noise.ConfidenceInterval
+	for curCountAlpha := increment; curCountAlpha < alpha; curCountAlpha += increment {
+		confInt, err := bm.computeConfidenceIntervalSplit(alpha, curCountAlpha)
+		if err != nil {
+			return noise.ConfidenceInterval{}, err
+		}
+		tightness := confInt.UpperBound - confInt.LowerBound
+		if tightness < tightest {
+			tightest = tightness
+			tightestConfInt = confInt
+		}
+	}
+	return tightestConfInt, nil
+}
+
+// computeConfidenceIntervalSplit computes a confidence interval for mean with a given 1 - countAlpha
+// confidence level for count. It computes the corresponding alpha for sum to return a confidence interval
+// with a confidence level of at least 1 - meanAlpha. The computation is based exclusively on the confidence
+// intervals for sum and count. Thus no privacy budget is consumed by this operation.
+//
+// Result() needs to be called before ComputeConfidenceInterval, otherwise this will return an error.
+func (bm *BoundedMeanFloat64) computeConfidenceIntervalSplit(meanAlpha, countAlpha float64) (noise.ConfidenceInterval, error) {
+	if !bm.resultReturned {
```

Comment:
I am not sure why we need this check here? This is a private method and the "main" method which calls it already has this check.

Reply:
Yep, you are right, this check is not needed.